### PR TITLE
[DEV-2642 & DEV-2631] Autocomplete City Search

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
Implements the city-search autocomplete functionality w/ a mock API.

**Technical details:**
Adds multiple new components depending on the `fetchCityAutocomplete` and `parseCities` methods on the `LocationPickerContainer`.

Also adds logic to the `createLocationObject` method so a location tag is created.

**JIRA Ticket:**
[DEV-2642](https://federal-spending-transparency.atlassian.net/browse/DEV-2642) & [DEV-2631](https://federal-spending-transparency.atlassian.net/browse/DEV-2631)

**Mockup:**
https://invis.io/RYA3XN5WP#/273832670_Homepage_2-2_E

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] Design review
- [x] [API contract](https://github.com/fedspendingtransparency/usaspending-api/pull/1804) merged (if applicable) PR merged 
- [x] back end PR merged (Not opened yet)
- [x] Verified cross-browser compatibility
- [x] locally test with live endpoint in DEV

### CODE REVIEW

**Comment by dlassobah** [comment]
Cannot fully vet the design without being able to see how it works with real data populating the dropdown/autocomplete, but based on what Max showed me earlier I am ok moving this forward. I will need to add an additional ticket to pursue fully implementing combo-boxes for the country/state fields (but de-scoped for this work) as indicated by the design. Please let me know when this prototype is able to pull a list of city data so I can see how it behaves

**Comment by MaxwellKendall** [comment]
@dlassobah will do.

**Comment by jonhill13** [comment]
@MaxwellKendall After viewing all the changes made to `src/js/components/search/filters/location/EntityDropdown.jsx ` to make it work as an autocomplete, I would move it into another component.

File: src/js/helpers/mapHelper.js [hunk 1]
```diff
@@ -475,3 +474,46 @@ export const performZIPGeocode = (zip) => {
         }
     };
 };
+
+export const getCitySearchRequestObj = (
```
Comment (jonhill13) [inline src/js/helpers/mapHelper.js]: @MaxwellKendall I would move this into the dataMapping folder. I would prefer the helpers to just be fetching.
You could do something like 
`
export const getCitySearchRequestObj = (searchText= "", limit= 40, filter) => {
  return {
    search_text: searchText,
    limit,
    filter: Object.assign({}, ...filter)
  }
};
`

or, 

`
import { pickBy, identity } = 'lodash';

export const getCitySearchRequestObj = (searchText= "", state='', country='', scope='') => {
  return pickBy({
    search_text: searchText,
    limit: 40,
    filter: {
                country_code: country,
                scope,
                state_code: state
            }
  }, identity)
};
`

**Comment by MaxwellKendall** [comment]
Got @jonhill13 's approval for above commits and also conducted follow up design review with @dlassobah and had help smoke testing the live api from @kbard.

Merging into dev.

**Pull request merged**